src/Cake.Common/Tools/VSTest/VSTestRunner.cs

```diff
@@ -152,37 +152,29 @@ protected override string GetToolName()
         /// <returns>The default tool path.</returns>
         protected override IEnumerable<FilePath> GetAlternativeToolPaths(VSTestSettings settings)
         {
-            foreach (var yearAndEdition in new[] { "2019/Enterprise", "2019/Professional", "2019/Community", "2017/Enterprise", "2017/Professional", "2017/Community" })
+            var vsRootRelativeToolPath = FilePath.FromString($"Common7/IDE/CommonExtensions/Microsoft/TestWindow/{VSTestConsoleExecutableName}");
+            foreach (var year in VisualStudio.Versions.TwentySeventeenAndLater)
             {
-                var path = GetYearAndEditionToolPath(yearAndEdition);
-                if (_fileSystem.Exist(path))
+                foreach (var edition in settings.AllowPreviewVersion
+                             ? VisualStudio.Editions.All
+                             : VisualStudio.Editions.Stable)
                 {
-                    yield return path;
+                    var path = VisualStudio.GetYearAndEditionToolPath(_environment, year, edition, vsRootRelativeToolPath);
+                    if (_fileSystem.Exist(path))
+                    {
+                        yield return path;
+                    }
                 }
             }
 
-            foreach (var version in new[] { "15.0", "14.0", "12.0", "11.0" })
+            foreach (var version in VisualStudio.Versions.TenToFourteen)
```

Comment:
Question: line here used to be eleven to fifteen, is this change by design?

Reply:
Interesting. I noticed the `15.0` in there, which would have probably never produced something useful. I did not notice that `10.0` was "missing" there (compared to the other instances of the number-versions). 
I'm guessing no harm in adding it here, right?